src/provider/cloud-element-templates/ConditionChecker.js

```diff
@@ -0,0 +1,94 @@
+import {
+  applyConditions
+} from './Condition';
+
+import CommandInterceptor from 'diagram-js/lib/command/CommandInterceptor';
+
+import { findOldProperty } from './cmd/ChangeElementTemplateHandler';
+import { setPropertyValue, unsetProperty } from './util/propertyUtil';
+
+/**
+ * Checks the conditions of an element template and sets/resets the
+ * corresponding properties on the element.
+ */
+export default class ConditionChecker extends CommandInterceptor {
+  constructor(eventBus, elementTemplates, commandStack, bpmnFactory) {
+    super(eventBus);
+
+    this._eventBus = eventBus;
+    this._elementTemplates = elementTemplates;
+    this._commandStack = commandStack;
+    this._bpmnFactory = bpmnFactory;
+
+    this.oldTemplate = null;
+
+    this.preExecute([
+      'element.updateProperties', 'element.updateModdleProperties'
+    ], this._saveConditionalState, true, this);
+
+    this.postExecute([
+      'element.updateProperties', 'element.updateModdleProperties', 'propertiesPanel.zeebe.changeTemplate'
+    ], this._applyConditions, true, this);
+  }
+
+  _saveConditionalState(context) {
+    const {
+      element
+    } = context;
+
+    const template = this._elementTemplates.get(element);
+
+    if (!template) {
+      return;
+    }
+
+    this.oldTemplate = applyConditions(element, template);
+  }
+
+  _applyConditions(context) {
+    const {
+      element
+    } = context;
+
+    const template = this._elementTemplates.get(element);
+
+    if (!template || !this.oldTemplate || template.id !== this.oldTemplate.id) {
```

Comment:
We should not store the old template in `this`, but rather store it in the `context`.

Handlers shall be stateless.